src/cmdstan/command.hpp

```diff
@@ -295,29 +211,84 @@ int command(int argc, const char *argv[]) {
+    int history_size
+        = get_arg_val<int_argument>(*pathfinder_arg, "history_size");
+    double init_alpha
+        = get_arg_val<real_argument>(*pathfinder_arg, "init_alpha");
+    double tol_obj = get_arg_val<real_argument>(*pathfinder_arg, "tol_obj");
+    double tol_rel_obj
+        = get_arg_val<real_argument>(*pathfinder_arg, "tol_rel_obj");
+    double tol_grad = get_arg_val<real_argument>(*pathfinder_arg, "tol_grad");
+    double tol_rel_grad
+        = get_arg_val<real_argument>(*pathfinder_arg, "tol_rel_grad");
+    double tol_param = get_arg_val<real_argument>(*pathfinder_arg, "tol_param");
+    int max_lbfgs_iters
+        = get_arg_val<int_argument>(*pathfinder_arg, "max_lbfgs_iters");
+    bool save_iterations
+        = get_arg_val<bool_argument>(*pathfinder_arg, "save_iterations");
+    int num_elbo_draws
+        = get_arg_val<int_argument>(*pathfinder_arg, "num_elbo_draws");
+    int num_draws = get_arg_val<int_argument>(*pathfinder_arg, "num_draws");
+    int num_psis_draws
+        = get_arg_val<int_argument>(*pathfinder_arg, "num_psis_draws");
+    int num_paths = get_arg_val<int_argument>(*pathfinder_arg, "num_paths");
+    if (num_paths == 1) {
+      return_code = stan::services::pathfinder::pathfinder_lbfgs_single<
+          false, stan::model::model_base>(
+          model, *(init_contexts[0]), random_seed, id, init_radius,
+          history_size, init_alpha, tol_obj, tol_rel_obj, tol_grad,
+          tol_rel_grad, tol_param, max_lbfgs_iters, num_elbo_draws, num_draws,
+          false, refresh, interrupt, logger, init_writer, sample_writers[0],
+          diagnostic_json_writers[0]);
+    } else {
+      std::string output_file
+          = get_arg_val<string_argument>(parser, "output", "file");
+      std::string pf_name;
+      std::string pf_suffix;
+      get_basename_suffix(output_file, pf_name, pf_suffix);
+      if (pf_suffix.empty())
+        pf_suffix = ".csv";
+      auto ofs = std::make_unique<std::ofstream>(pf_name + "_pathfinder"
+                                                 + pf_suffix);
+      if (sig_figs > -1)
+        ofs->precision(sig_figs);
+      stan::callbacks::unique_stream_writer<std::ofstream> pathfinder_writer(
+          std::move(ofs), "# ");
+      write_stan(pathfinder_writer);
+      write_model(pathfinder_writer, model.model_name());
+      write_datetime(pathfinder_writer);
+      parser.print(pathfinder_writer);
+      return_code = stan::services::pathfinder::pathfinder_lbfgs_multi<
+          stan::model::model_base>(
+          model, init_contexts, random_seed, id, init_radius, history_size,
+          init_alpha, tol_obj, tol_rel_obj, tol_grad, tol_rel_grad, tol_param,
+          max_lbfgs_iters, num_elbo_draws, num_draws, num_psis_draws, num_paths,
+          true, refresh, interrupt, logger, init_writers, sample_writers,
+          diagnostic_json_writers, pathfinder_writer,
+          diagnostic_json_writers[0]);
```

Comment:
Is this passing the same JSON diagnostic writer as both the writer for the first pathfinder and for the overall multi-pathfinder?

Reply:
no.   diagnostic writers are only on the individual single-path pathfinders.  the overall multi-pathfinder writer is the importance resampled draws from the single-path pathfinders.  no gradients involved.

Reply:
added dummy_writer.